### ISSUE TITLE
Incremental variance estimation instead of smoothing in batch normalization 

### ISSUE THREAD

**Issue by tlkvstepan** [pytorch/pytorch#1962]
I think it might be nice to estimate variances using incremental algorithm instead of smoothing variances estimated in every batch. 
* In this case BN will work fine even with batch size of 1.
* Estimated variance will be less biased when batch size is small.

**Comment by soumith** [comment]
at this point we will likely not change the behavior of BatchNorm, there's too much code that depends on it. Practically, you do run the estimation over an epoch of training and that's large enough to debias the estimate enough.
